app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -444,11 +444,19 @@ def smart_proxy_service
 
       def rhsm_url
         if pulp_primary?
-          "https://#{URI.parse(url).host}/rhsm"
+          URI("https://#{URI.parse(url).host}/rhsm")
```

Comment:
On my katello-devel-setup
```
proxy = SmartProxy.find(1)

proxy.setting(SmartProxy::PULP3_FEATURE, 'rhsm_url')
# => "https://localhost/rhsm"

proxy.rhsm_url
# => "https://centos7-katello-devel-stable.example.com/rhsm"
```

Is the value of `rhsm_url` in proxy plugin set by installer or users have to do it manually?
Because if they have to set it manually, there is chance that they didn't do it and it will break the workflow.

Reply:
It will be set by the installer (https://github.com/theforeman/puppet-foreman_proxy_content/pull/390) but that hasn't been merged yet. You will also need a fallback for n-1 support. Foreman Proxy 3.0 won't have it and it's supported to register a 3.0 Proxy with a 3.1 Foreman. This fallback is the current code.

Reply:
It's set by the installer as of https://github.com/theforeman/puppet-foreman_proxy_content/pull/384 but the apache vhost to proxy /rhsm on 443 is deployed in https://github.com/theforeman/puppet-foreman_proxy_content/pull/390

Reply:
Since it should land in 4.3 I opened https://github.com/Katello/katello/pull/9777 to implement this.